src/Plugin/Deriver/EntityAliasTypeDeriver.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\pathauto\Plugin\Deriver\EntityAliasTypeDeriver.
+ */
+
+namespace Drupal\pathauto\Plugin\Deriver;
+
+use Drupal\Core\Plugin\Context\ContextDefinition;
+use Drupal\ctools\Plugin\Deriver\EntityDeriverBase;
+
+class EntityAliasTypeDeriver extends EntityDeriverBase {
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    foreach ($this->entityManager->getDefinitions() as $entity_type_id => $entity_type) {
+      if ($entity_type->hasLinkTemplate('canonical')) {
+        $this->derivatives[$entity_type_id] = $base_plugin_definition;
+        $this->derivatives[$entity_type_id]['label'] = $this->t('@label', ['@label' => $entity_type->getLabel()]);
```

Comment:
I don't think t('@label') makes sense? getlabel() returns a translation object too, so that works just fine now.